src/main/java/com/dnd/namuiwiki/domain/survey/model/dto/GetSurveyResponse.java

```diff
@@ -0,0 +1,61 @@
+package com.dnd.namuiwiki.domain.survey.model.dto;
+
+import com.dnd.namuiwiki.common.exception.ApplicationErrorException;
+import com.dnd.namuiwiki.common.exception.ApplicationErrorType;
+import com.dnd.namuiwiki.domain.question.entity.Question;
+import com.dnd.namuiwiki.domain.survey.model.entity.Survey;
+import com.dnd.namuiwiki.domain.survey.type.Period;
+import com.dnd.namuiwiki.domain.survey.type.Relation;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+@AllArgsConstructor
+public class GetSurveyResponse {
+    private String senderName;
+    private Period period;
+    private Relation relation;
+    private LocalDateTime createdAt;
+    private List<SingleQuestionAndAnswer> questionAndAnswers;
+
+    @Getter
+    @AllArgsConstructor
+    private static class SingleQuestionAndAnswer {
+        private String questionTitle;
+        private Object answer;
+        private String reason;
+
+        static SingleQuestionAndAnswer from(String questionTitle, Survey.Answer surveyAnswer) {
+            return new SingleQuestionAndAnswer(questionTitle, surveyAnswer.getAnswer(), surveyAnswer.getReason());
+        }
+    }
+
+    public static GetSurveyResponse from(Survey survey, List<Question> questions) {
+        var singleQuestionAndAnswers = pairQuestionAndAnswer(survey, questions);
```

Comment:
```suggestion
        List<SingleQuestionAndAnswer> singleQuestionAndAnswers = survey.getAnswers().stream().map(answer -> {
            Question question = answer.getQuestion();
            return SingleQuestionAndAnswer.from(question.getTitle(), answer);
        }).toList();
```

혹시 이 코드와 다른 점이 있을까요?
Survey.Answer에서 답변에 대한 Question을 가지고 있어서 따로 question을 조회할 필요 없어 보여서요!

Reply:
Answer 내의 question에서 question title을 조회하기 위해 매번 새로운 query가 발생할 것을 고려했습니다.(question에 LAZY 옵션이 걸려있기 때문) 따라서 question을 모두 불러온 뒤 question과 answer를 pairing 하는 방식으로 구현했습니다.
+ question 같은 경우 자주 조회되고 변할 가능성이 낮은 데이터라 캐시에 올려두는게 좋을 것 같습니다